app/Jobs/UpdateWikiSiteStatsJob.php

```diff
@@ -74,7 +75,8 @@ private function updateSiteStats (Wiki $wiki): void
                 $update[$field] = $value;
             }
         }
-
-        $wiki->wikiSiteStats()->updateOrCreate(['wiki_id' => $wiki->id], $update);
+        DB::transaction(function () use ($wiki, $update) {
+            $wiki->wikiSiteStats()->lockForUpdate()->updateOrCreate(['wiki_id' => $wiki->id], $update);
```

Comment:
praise: Looks good; I read up a little on lockForUpdate and it look like it would do what we need it to